code/modules/cm_aliens/XenoStructures.dm

```diff
@@ -844,6 +845,227 @@
+			return candidate.client
+	// Then observers
+	var/list/observer_list_copy = shuffle(get_alien_candidates(hive))
+
+	for(var/mob/candidate in observer_list_copy)
+		if(try_roll_candidate(hive, candidate, playtime_restricted = TRUE))
+			return candidate.client
+	// Lastly all of the above again, without playtime requirements
+	for(var/mob/living/carbon/xenomorph/candidate in total_xenos_copy)
+		if(try_roll_candidate(hive, candidate, playtime_restricted = FALSE))
+			return candidate.client
+	for(var/mob/candidate in observer_list_copy)
+		if(try_roll_candidate(hive, candidate, playtime_restricted = FALSE))
+			return candidate.client
+	message_admins("Failed to find a client for the King, releasing as freed mob.")
+	return null
+
+/obj/effect/alien/resin/king_cocoon/proc/choose_candidate(expedite = FALSE)
+	rolling_candidates = TRUE
+	chosen_candidate = roll_candidates()
+	rolling_candidates = FALSE
+	
+	if(expedite)
+		animate_hatch_king()
+		return
+	
+	marine_announcement("ALERT.\n\nUNSUAL ENERGY BUILDUP DETECTED IN [get_area_name(loc)].\n\nESTIMATED TIME UNTIL COMPLETION - 1 MINUTES.", "[MAIN_AI_SYSTEM] Biological Scanner", 'sound/misc/notice1.ogg')
+	var/datum/hive_status/hive
+	for(var/cur_hive_num in GLOB.hive_datum)
+		hive = GLOB.hive_datum[cur_hive_num]
+		if(!length(hive.totalXenos))
+			continue
+		if(cur_hive_num == hive_number)
+			xeno_announcement(SPAN_XENOANNOUNCE("The King will hatch in approximately one minute."), cur_hive_num, XENO_GENERAL_ANNOUNCE)
+		else
+			xeno_announcement(SPAN_XENOANNOUNCE("Another hive's King will hatch in approximately one minute."), cur_hive_num, XENO_GENERAL_ANNOUNCE)
+
+	// Technically slow players will delay us so its not just 10 minutes
+	timer = addtimer(CALLBACK(src, PROC_REF(animate_hatch_king)), 1 MINUTES, TIMER_UNIQUE|TIMER_STOPPABLE|TIMER_DELETE_ME)
+	
+
+
+/obj/effect/alien/resin/king_cocoon/proc/animate_hatch_king()
+	flick("hatching", src)
+	addtimer(CALLBACK(src, PROC_REF(hatch_king)), 2 SECONDS, TIMER_UNIQUE|TIMER_STOPPABLE)
+
+	marine_announcement("ALERT.\n\nEXTREME ENERGY INFLUX DETECTED IN [get_area_name(loc)].\n\nCAUTION IS ADVISED.", "[MAIN_AI_SYSTEM] Biological Scanner", 'sound/misc/notice1.ogg')
+	var/datum/hive_status/hive
+	for(var/cur_hive_num in GLOB.hive_datum)
+		hive = GLOB.hive_datum[cur_hive_num]
+		if(!length(hive.totalXenos))
+			continue
+		if(cur_hive_num == hive_number)
+			xeno_announcement(SPAN_XENOANNOUNCE("All hail the King."), cur_hive_num, XENO_GENERAL_ANNOUNCE)
+		else
+			xeno_announcement(SPAN_XENOANNOUNCE("Another hive's King has hatched!"), cur_hive_num, XENO_GENERAL_ANNOUNCE)
+
+/obj/effect/alien/resin/king_cocoon/proc/hatch_king()
+	icon_state = "hatched"
+	hatched = TRUE
+
+	QDEL_LIST(blockers)
+
+	var/mob/living/carbon/xenomorph/king/king = new(get_turf(src))
+	if(chosen_candidate?.mob)
+		var/mob/old_mob = chosen_candidate.mob
+		old_mob.mind.transfer_to(king)
+		old_mob.free_for_ghosts(TRUE)
```

Comment:
We need to test `isliving(old_mob)` and the `old_mob.stat` of the old_mob to ensure we don't free something dead.